### ISSUE TITLE
Config still points to removed wallet on removed key[BUG] 

### ISSUE THREAD

**Issue by coderasm** [Chia-Network/chia-blockchain#3141]
**Describe the bug**
If I remove a key, the wallet attached to that key is still referenced for the farmer xch_target_address and pool xch_target_address.

**To Reproduce**

1. run `chia keys generate`
2. run `chia keys delete_all`
3. run `chia keys add`
4. enter a mnemonic different from the mnemonic of the first removed key
5. the farmer xch_target_address and pool xch_target_address will still reference the wallet address attached to the first removed key

**Expected behavior**
The wallet address of the most recently added key will be used for the farmer xch_target_address and pool xch_target_address

**Screenshots**
![chia_key_remove_bug](https://user-images.githubusercontent.com/446396/116633219-cdf53400-a90d-11eb-99ca-2a7de3e0a749.PNG)

**Desktop (please complete the following information):**

- OS: ubuntu
- OS Version/Flavor: server 20.04
- CPU: r9-5900x

**Additional context**
Attached is a screenshot of my bash history of commands. I lost 4 chia to this problem. I assumed the new key was tied to my already existing mnemonic. I did not know a new mnemonic was created for a key. I used the `chia keys generate` command which automatically adds the new key and does not display the mnemonic. Had I used the `chia keys generate_and_print` command, I would have seen the mnemonic and saved it off. I also did not know a newly generated wallet address attached to a new key would automatically be inserted in the config as the farmer and pool target addresses. I also assumed deleting all keys and adding only my original key from my first mnemonic would set all configuration to use my last added key. That did not happen since I was farming for almost two weeks to a wallet I no longer had the private key for since I deleted it without backing up the mnemonic.

**Comment by doma2345** [comment]
i lost my first two chia farmed to this bug

**Comment by xbirt** [comment]
I wouldn't describe this as a bug but rather as a missing feature. The xch_target_address should never be changed implicitly.

It would however be useful to add some code that asks the user if they want to also change the xch_target_address in this scenario.

**Comment by SashoBushido** [comment]
Same issue, guys. Had farmed solo for a decent amount of time and then I switched to a pool. On the pool page I saw that I have recently found a block, and then couldn't see the 0.25xch reward in my wallet. After performing a block search in [https://www.chiaexplorer.com/](https://www.chiaexplorer.com/) I found that besides this one, I had earned a block back at the time when I farmed solo. So 2.25xch have vanished. And that happened because I created a test wallet before the actual one when I was getting familiar with the GUI.
